src/test/java/reposense/model/RepoLocationTest.java

```diff
@@ -0,0 +1,71 @@
+package reposense.model;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import reposense.parser.InvalidLocationException;
+import reposense.util.AssertUtil;
+
+public class RepoLocationTest {
+    private static final String TEST_REPO_BETA = "https://github.com/reposense/testrepo-Beta.git";
+    private static final String TEST_REPO_DELTA = "https://github.com/reposense/testrepo-Delta.git";
+
+    @Test
+    public void repoLocation_validRepoUrl_success() throws Exception {
+        assertValidLocation(TEST_REPO_BETA, "testrepo-Beta", "reposense");
+        assertValidLocation(TEST_REPO_DELTA, "testrepo-Delta", "reposense");
+    }
+
+    @Test
+    public void repoLocation_invalidRepoUrl_throwsInvalidLocationException() {
+        // ftp url should be rejected
+        assertInvalidLocation("ftp://github.com/reposense/testrepo-Delta.git");
+        // non GitHub url should rejected
+        assertInvalidLocation("https://gitlab.com/reposense/RepoSense.git");
+        // url without organisation name should be rejected
+        assertInvalidLocation("https://github.com/reposense.git");
+        // url without repo name should be rejected
+        assertInvalidLocation("https://github.com/reposense/.git");
+        assertInvalidLocation("https://github.com/reposense");
+        // url without a .git suffix should be rejected
+        assertInvalidLocation("https://github.com/reposense/RepoSensegit");
+        assertInvalidLocation("https://github.com/reposense/RepoSense-git");
+        assertInvalidLocation("https://github.com/reposense/RepoSense.gi");
+        assertInvalidLocation("https://github.com/reposense/RepoSense");
+    }
+
+    @Test
+    public void repoLocation_invalidBranchUrl_throwsInvalidLocationException() {
+        // ftp url should be rejected
+        assertInvalidLocation("ftp://github.com/reposense/RepoSense/tree/feature_branch_issue#1010");
+        // url without branch name should be rejected
+        assertInvalidLocation("https://github.com/reposense/RepoSense/tree/");
+        assertInvalidLocation("https://github.com/reposense/RepoSense/tree");
+        // url without the 'tree' keyword should be rejected
+        assertInvalidLocation("https://github.com/reposense/RepoSense/feature_branch_issue#1010");
+        assertInvalidLocation("https://github.com/reposense/RepoSense/TREE/feature_branch_issue#1010");
+        assertInvalidLocation("https://github.com/reposense/RepoSense/tre/feature_branch_issue#1010");
```

Comment:
Is `https://github.com/reposense/RepoSense/tree/master` considered a valid URL? If yes, do add it to the successful tests. Otherwise, it should be added as an invalid URL test and the comments should be updated.